x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/FrozenCacheService.java

```diff
@@ -67,9 +79,48 @@
         Setting.Property.NodeScope
     );
 
-    public static final Setting<ByteSizeValue> SNAPSHOT_CACHE_SIZE_SETTING = Setting.byteSizeSetting(
+    public static final Setting<ByteSizeValue> SNAPSHOT_CACHE_SIZE_SETTING = new Setting<>(
         SHARED_CACHE_SETTINGS_PREFIX + "size",
-        ByteSizeValue.ZERO,
+        ByteSizeValue.ZERO.getStringRep(),
+        s -> ByteSizeValue.parseBytesSizeValue(s, SHARED_CACHE_SETTINGS_PREFIX + "size"),
+        new Setting.Validator<ByteSizeValue>() {
+
+            @Override
+            public void validate(final ByteSizeValue value) {
+
+            }
+
+            @Override
+            public void validate(final ByteSizeValue value, final Map<Setting<?>, Object> settings) {
+                if (value.getBytes() > 0) {
+                    @SuppressWarnings("unchecked") final List<DiscoveryNodeRole> roles =
+                        (List<DiscoveryNodeRole>) settings.get(NodeRoleSettings.NODE_ROLES_SETTING);
+                    final DiscoveryNode fake = new DiscoveryNode(
```

Comment:
Rather than constructing this fake `DiscoveryNode`, what about making `DataTier.isFrozenNode(...)` invoke a separate static method that takes a `Set<DiscoveryNodeRole>` so that method could be used directly? It feels weird to construct a whole fake node just for this purpose.

Reply:
It felt weird to add a whole method just for this purpose. 🤷‍♀️

Especially since it's hard to come up with a good name for the method. `isFrozenNode` isn't great because it's not testing a node, `hasFrozenRole` isn't great because it can also include the `data` role and should return true.

I pushed 892f575e1234dc905cf67a1316ad24c738abca4e.

Reply:
It felt weird to add a whole method just for this purpose. 🤷‍♀️

Especially since it's hard to come up with a good name for the method. `isFrozenNode` isn't great because it's not testing a node, `hasFrozenRole` isn't great because it can also include the `data` role and should return true.

I pushed 892f575e1234dc905cf67a1316ad24c738abca4e.